app/controllers/DigitalSubscription.scala

```diff
@@ -138,10 +145,13 @@ class DigitalSubscription(
     result.fold(
       { error =>
         SafeLogger.error(scrub"[${request.uuid}] Failed to create new $billingPeriod Digital Subscription, due to $error")
-        InternalServerError
+        error match {
+          case "validation-failure" => BadRequest
```

Comment:
This is a cry for help.

Reply:
I tried to use types like a sane person, but the eitherT/covariance confusion is too much. Do you have time to chat with me about this @jacobwinch ?